src/configuration/OpenSearchDetails.tsx

```diff
@@ -18,14 +20,73 @@ const indexPatternTypes = [
 type Props = {
   value: DataSourceSettings<OpenSearchOptions>;
   onChange: (value: DataSourceSettings<OpenSearchOptions>) => void;
+  saveOptions: () => Promise<void>;
+  datasource: OpenSearchDatasource;
 };
 export const OpenSearchDetails = (props: Props) => {
-  const { value, onChange } = props;
+  const { value, onChange, saveOptions, datasource } = props;
+  const [versionErr, setVersionErr] = useState<string>('');
+
+  const setVersion = async () => {
+    setVersionErr('');
+    await saveOptions();
+    try {
+      const version = await datasource.getOpenSearchVersion();
```

Comment:
can `getOpenSearchVersion` check for supported versions and throw an error if it's >=7.11.0? right now it can successfully set a version and it appears to be ok, but i don't see an error until i save and test the datasource.

Reply:
done!